### PR TITLE
Langchain Integration

### PR DESCRIPTION
**Pull request by anaaaiva**

### CODE REVIEW

File: src/app.py [hunk 1]
```diff
@@ -1,34 +1,97 @@
...
-)
+from data_loading import load_data_test
+
+# from utils import (
+#     build_index,
+#     generate_answer,
+#     generate_embedding,
+#     load_data,
+#     query_index,
+# )
```
Comment (qw1zzard) [inline src/app.py]: Remove unused code (and from utils, too)

File: src/app.py [hunk 2]
```diff
@@ -1,34 +1,97 @@
...
+            # if query_embedding.size > 0:
+            #     with st.spinner('Fetching relevant context...'):
+            #         context = query_index(index, query_embedding, documents)
+
+            #     with st.spinner('Generating answer...'):
+            #         answer = generate_answer(context, query)
+
+            #     st.write(answer)
+            # else:
+            #     st.error('Failed to generate embedding for the query.')
```
Comment (qw1zzard) [inline src/app.py]: And here too

File: src/app.py (hunk 1 above)
Comment (anaaaiva) [reply src/app.py]: resolved

File: src/app.py (hunk 2 above)
Comment (anaaaiva) [reply src/app.py]: resolved

**Review by qw1zzard — APPROVED** [review]
lgtm

**Pull request merged**